flashinfer/experimental/sm12x/attention/_shared/cute/pipeline.py

```
@@ -0,0 +1,101 @@
+            self._phase_index ^= 1
+        else:
+            self._phase_index += 1
+
+    def __extract_mlir_values__(self):
+        return [self._phase_index.ir_value()]
+
+    def __new_from_mlir_values__(self, values):
+        return PipelineStateSimple(self.stages, Int32(values[0]))
+
+
+def make_pipeline_state(type: PipelineUserType, stages: int):
```

Ruff (0.15.21)

[error] Function argument `type` is shadowing a Python builtin

(A002)